### DISCUSSION TITLE
cpg0038-tegtmeyer-neuropainting

### DISCUSSION THREAD

**Discussion by shntnu** [broadinstitute/cellpainting-gallery#94] [Dataset descriptions]
We will follow the process documented here https://github.com/broadinstitute/cellpainting-gallery/wiki/Data-Transfer-Workflow

---

Checklist from https://github.com/broadinstitute/cellpainting-gallery/wiki/Data-Transfer-Workflow

Champion: Shantanu SIngh
Dataset name: `cpg0038-tegtmeyer-neuropainting `

- [x] Contact a maintainer asking to deposit data, following the instructions in the [contribution docs](https://broadinstitute.github.io/cellpainting-gallery/contributing_to_cpg.html#contributing-data-to-the-gallery).
- [x] The Maintainer will assign a dataset identifier (e.g., `cpg0031`) by incrementing the identifier of the most recent dataset in the dataset [list](https://github.com/broadinstitute/cellpainting-gallery/blob/main/documentation/prefixes.md), and update the list with the new identifier.
- [x] The Maintainer will create a new [discussion](https://github.com/broadinstitute/cellpainting-gallery/discussions) using the dataset identifier as the title, which will be used to track the progress of this workflow
- [x] The Maintainer will identify an internal Champion for the project.
- [x] The Champion and the Contributor will discuss the structure of the data, ensuring it follows the guidelines discussed in the [contribution docs](https://broadinstitute.github.io/cellpainting-gallery/contributing_to_cpg.html#contributing-data-to-the-gallery).
- [x] The Champion and the Contributor will come up with an appropriate name for the dataset (e.g., `cpg0031-caicedo-cmvip`).
- [x] The Champion will request credentials for the Contributor by specifying the prefix (email [@leoank](https://github.com/leoank) and cc the Contributor). Each credential is for a single prefix on the staging bucket; multiple people can assume that same credential.
- [x] The Contributor will use provided credentials to upload data to the staging bucket, following [instructions](https://github.com/broadinstitute/cytoskel/blob/main/cytoskel/docs/access_cpg_staging.md) on accessing and using the credentials.
- [x] The Champion will verify data structure and give the go-ahead to transfer from the staging bucket to the production bucket.
- [x] A Maintainer will perform the transfer using an automated process with AWS S3 Batch.
- [x] A Maintainer will disable the staging prefix, delete the data from staging, and delete credentials.

**Comment by shntnu** [comment]
Data transfer is in process. Note that some batches have been renamed, for clarity

```sh
SOURCE_DIR=s3://imaging-platform/projects/2019_05_28_Neuronal_Cell_Painting
DEST_DIR=s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images

s5cmd --log error sync "${SOURCE_DIR}/NCP_STEM_1/*" ${DEST_DIR}/NCP_STEM_1/
s5cmd --log error sync "${SOURCE_DIR}/2022_03_03_NCP_NEURONS_2_20x/*" ${DEST_DIR}/NCP_NEURONS_2_20x/ # DONE: run again once objects restored
s5cmd --log error sync "${SOURCE_DIR}/2022_03_03_NCP_NEURONS_2_63x/*" ${DEST_DIR}/NCP_NEURONS_2_63x/ # DONE: run again once objects restored
s5cmd --log error sync "${SOURCE_DIR}/2022-08-07_batch/*" ${DEST_DIR}/NCP_ASTROCYTES_1/
s5cmd --log error sync "${SOURCE_DIR}/NCP_PROGENITORS_1/illum/BR00127194/*" ${DEST_DIR}/NCP_PROGENITORS_1/illum/BR00127194/
s5cmd --log error sync "${SOURCE_DIR}/NCP_PROGENITORS_1/images/BR00127194__2021-09-03T17_06_45-Measurement_2/*" ${DEST_DIR}/NCP_PROGENITORS_1/images/BR00127194__2021-09-03T17_06_45-Measurement_2/

SOURCE_DIR=s3://imaging-platform/projects/2019_05_28_Neuronal_Cell_Painting/workspace/load_data_csv
DEST_DIR=s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/workspace/load_data_csv
s5cmd --log error sync "${SOURCE_DIR}/NCP_STEM_1/*" ${DEST_DIR}/NCP_STEM_1/
s5cmd --log error sync "${SOURCE_DIR}/2022_03_03_NCP_NEURONS_2_20x/*" ${DEST_DIR}/NCP_NEURONS_2_20x/
s5cmd --log error sync "${SOURCE_DIR}/2022_03_03_NCP_NEURONS_2_63x/*" ${DEST_DIR}/NCP_NEURONS_2_63x/
s5cmd --log error sync "${SOURCE_DIR}/2022-08-07_batch/*" ${DEST_DIR}/NCP_ASTROCYTES_1/
s5cmd --log error sync "${SOURCE_DIR}/NCP_PROGENITORS_1/BR00127194/*" ${DEST_DIR}/NCP_PROGENITORS_1/BR00127194/

# update paths in load data csv files
find . -type f -name "*.csv" -exec sed -i '' 's/2022_03_03_NCP_NEURONS_2_20x/NCP_NEURONS_2_20x/g' {} +
find . -type f -name "*.csv" -exec sed -i '' 's/2022_03_03_NCP_NEURONS_2_63x/NCP_NEURONS_2_63x/g' {} +
find . -type f -name "*.csv" -exec sed -i '' 's/2022-08-07_batch/NCP_ASTROCYTES_1/g' {} +


```

**Reply by shntnu** [reply]
6 plates:

https://lite.datasette.io/?parquet=https%3A%2F%2Fmedia.githubusercontent.com%2Fmedia%2Fbroadinstitute%2FNeuroPainting%2Frefs%2Fheads%2Fmaster%2F5.reanalysis%2Foutput%2Fprocessed%2Fvar_mad_int_featselect%2Fcombined.parquet#/data?sql=SELECT+%0A++++Metadata_Plate%2C+%0A++++Metadata_cell_type%2C+%0A++++Metadata_plating_density%2C+%0A++++COUNT%28*%29+AS+count%0AFROM+combined%0AGROUP+BY+%0A++++Metadata_Plate%2C+%0A++++Metadata_cell_type%2C+%0A++++Metadata_plating_density%3B

<img width="390" alt="image" src="https://github.com/user-attachments/assets/5e5c8164-a4bc-4790-87c9-969aa5d22395" />


Note that neuron plates were imaged in 63x as well as 20x

images:

<img width="459" alt="image" src="https://github.com/user-attachments/assets/bac0b980-f9bc-4c51-8885-44760d5e026a" />

load_data_csv (Paths need to be updated)

<img width="293" alt="image" src="https://github.com/user-attachments/assets/0cfb9230-cb20-4d2e-9fee-5915aba2ebd3" />

Metadata

<img width="302" alt="image" src="https://github.com/user-attachments/assets/0a8757e3-e84b-4158-9f72-eb2cfb6b0543" />

**Reply by shntnu** [reply]
Pipeline files

Progenitors:
https://github.com/broadinstitute/NeuroPainting/tree/master/1.run-workflows/pipeline/NCP_PROGENITORS_1

Neurons:
https://github.com/broadinstitute/NeuroPainting/tree/master/1.run-workflows/pipeline/2022_03_03_NCP_NEURONS_2_20x
https://github.com/broadinstitute/NeuroPainting/tree/master/1.run-workflows/pipeline/2022_03_03_NCP_NEURONS_2_63x

Stem cells:
https://github.com/broadinstitute/imaging-platform-pipelines/tree/master/cellpainting_ipsc_20x_phenix_with_bf_bin1_cp3

Astrocytes:
https://imaging-platform.s3.amazonaws.com/projects/2019_05_28_Neuronal_Cell_Painting/workspace/pipelines/2022-08-07_batch

I've added them to the pipelines repo via this PR: https://github.com/broadinstitute/imaging-platform-pipelines/pull/32

**Reply by ErinWeisbart** [reply]
Synced into CPG with:

<details>
<summary>pipeline sync</summary>
<br>

```bash
aws s3 sync /Users/eweisbar/Desktop/github/imaging-platform-pipelines/cellpainting_neuralprogenitor_20x_phenix s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/workspace/pipelines/NCP_PROGENITORS_1/ --profile CPGnew

aws s3 sync /Users/eweisbar/Desktop/github/imaging-platform-pipelines/cellpainting_astrocyte_20x_phenix s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/workspace/pipelines/NCP_ASTROCYTES_1/ --profile CPGnew

aws s3 sync /Users/eweisbar/Desktop/github/imaging-platform-pipelines/cellpainting_neuron_20x_phenix s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/workspace/pipelines/NCP_NEURONS_2_20x/ --profile CPGnew

aws s3 sync /Users/eweisbar/Desktop/github/imaging-platform-pipelines/cellpainting_neuron_63x_phenix s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/workspace/pipelines/NCP_NEURONS_2_63x/ --profile CPGnew

aws s3 sync /Users/eweisbar/Desktop/github/imaging-platform-pipelines/cellpainting_ipsc_20x_phenix_with_bf_bin1_cp3 s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/workspace/pipelines/NCP_STEM_1/ --profile CPGnew
```
</details>

**Comment by ErinWeisbart** [comment]
Our historic internal data structure for images that undergo projection is different than that of the CPG. (In CPG, raw images always go in `images` and projected images go to `images_projected` while internally we put images to `images_unprojected` and then once projected to `images`). So I corrected image location for 3 batches that had z-stacks with:
```
aws s3 mv --recursive s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_ASTROCYTES_1/images/ s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_ASTROCYTES_1/images_projected/ 

aws s3 mv --recursive s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_20x/images/ s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_20x/images_projected/ 

aws s3 mv --recursive s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_63x/images/ s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_63x/images_projected/ 

```
and once those were done:
```
aws s3 mv --recursive s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_ASTROCYTES_1/images_unprojected/ s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_ASTROCYTES_1/images/ 

aws s3 mv --recursive s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_20x/images_unprojected/ s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_20x/images/ 

aws s3 mv --recursive s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_63x/images_unprojected/ s3://cellpainting-gallery/cpg0038-tegtmeyer-neuropainting/broad/images/NCP_NEURONS_2_63x/images/
```

**Comment by ErinWeisbart** [comment]
Corrected file paths to match CPG structure with code below. Note that it outputs two versions - one to go into CPG that is compliant with CPG structure and one I uploaded to our private bucket that can be used for DCP.

<details>
<summary>load_data_csv correction</summary>
<br>

```python
import pandas as pd
import os

root = '/Users/eweisbar/Desktop/load_data_csv_orig'
output = '/Users/eweisbar/Desktop/tegtmeyer_cpg'
output_run = '/Users/eweisbar/Desktop/tegtmeyer_run'

id = 'cpg0038-tegtmeyer-neuropainting'
source = 'broad'

zbatches = ['NCP_ASTROCYTES_1','NCP_NEURONS_2_20x','NCP_NEURONS_2_63x']

def correct(df, id,source,batch,plate):
    cols = df.columns
    for suffix in [x.split('_',1)[1] for x in cols if 'FileName' in x]:
        assert len([x.split('_',1)[1] for x in cols if 'FileName' in x]) == len(set([x.split('_',1)[1] for x in cols if 'FileName' in x])), f"{id} {source} {batch} {plate} {df} failed"
        if df[f'PathName_{suffix}'][0].endswith('/'):
            df[f'URL_{suffix}'] = df[f'PathName_{suffix}'] + df[f'FileName_{suffix}']
        else:
            df[f'URL_{suffix}'] = df[f'PathName_{suffix}'] + '/' + df[f'FileName_{suffix}']
        df.drop(columns=[f'PathName_{suffix}',f'FileName_{suffix}'],inplace=True)
        # move URL columns to the front of the dataframe
        col = df.pop(f'URL_{suffix}')
        df.insert(0, col.name, col)
    return df


for batch in [x for x in os.listdir(root) if x != '.DS_Store']:
    
        for plate in [x for x in os.listdir(os.path.join(root,batch)) if x != '.DS_Store']:
            if not os.path.exists(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate)):
                os.makedirs(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate), exist_ok=True)
            if not os.path.exists(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate)):
                os.makedirs(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate), exist_ok=True)

            ld = pd.read_csv(os.path.join(root,batch,plate,'load_data.csv'))
            ld_illum = pd.read_csv(os.path.join(root,batch,plate,'load_data_with_illum.csv'))
            
            if batch in zbatches:
                ld_un = pd.read_csv(os.path.join(root,batch,plate,'load_data_unprojected.csv'))

                for df in [ld,ld_illum,ld_un]:
                    df.replace('projects/2019_05_28_Neuronal_Cell_Painting',f'{id}/{source}/images',regex=True,inplace=True)
                for df in [ld, ld_illum]:
                    df.replace(f'{batch}/images/',f'{batch}/images_projected/',regex=True,inplace=True)
                ld_un.replace('images_unprojected','images',regex=True,inplace=True)

                ld.to_csv(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate,'load_data_projected.csv'), index=False)
                ld_illum.to_csv(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate,'load_data_projected_with_illum.csv'), index=False)
                ld_un.to_csv(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate,'load_data.csv'), index=False)

                for df in [ld,ld_illum,ld_un]:
                    df.replace('/home/ubuntu/bucket/',f's3://cellpainting-gallery/',regex=True,inplace=True)
                    df = correct(df,id, source, batch, plate)

                ld.to_csv(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate,'load_data_projected.csv'), index=False)
                ld_illum.to_csv(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate,'load_data_projected_with_illum.csv'), index=False)
                ld_un.to_csv(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate,'load_data.csv'), index=False)
            else:
                for df in [ld,ld_illum]:
                    df.replace('projects/2019_05_28_Neuronal_Cell_Painting',f'{id}/{source}/images',regex=True,inplace=True)

                ld.to_csv(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate,'load_data.csv'), index=False)
                ld_illum.to_csv(os.path.join(output_run,id,source,'workspace','load_data_csv',batch,plate,'load_data_with_illum.csv'), index=False)

                for df in [ld,ld_illum]:
                    df.replace('/home/ubuntu/bucket/',f's3://cellpainting-gallery/',regex=True,inplace=True)
                    df = correct(df,id, source, batch, plate)

                ld.to_csv(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate,'load_data.csv'), index=False)
                ld_illum.to_csv(os.path.join(output,id,source,'workspace','load_data_csv',batch,plate,'load_data_with_illum.csv'), index=False)
```
</details>

**Reply by shntnu** [reply]
<img width="340" alt="image" src="https://github.com/user-attachments/assets/102bc310-fed2-4ceb-a218-16ae44d19b59" />

@ErinWeisbart Should these be moved to be under projects? (actually, if so, it should be moved to be under `2019_05_28_Neuronal_Cell_Painting`)

**Reply by ErinWeisbart** [reply]
I would like for them to be where they are, but they only need to be there temporarily. They are the same information as what is now in the gallery, just with the appropriate structuring for being read by Distributed-CellProfiler. So they will be in the imaging-platform bucket only until I have finished the (re-)analysis and then I will delete them from the bucket. (DCP can't read the format we standardized on for the gallery right now but it's much easier if the files are structured as if they were on the gallery)

**Reply by shntnu** [reply]
Thanks for clarifying, all set!

**Comment by shntnu** [comment]
@ErinWeisbart said:

> I've added [a draft PR](https://github.com/broadinstitute/cellpainting-gallery/pull/137) to add this to README since it has a preprint. There are a number of /workspace folders missing (analysis, backend, profiles). Would you like for me to sync them over?

@shntnu said:

> This is a tricky one because there are analysis, backend, profiles that you [created](https://github.com/broadinstitute/2016_08_01_RadialMitochondriaDistribution_donna/issues/28) recently (for the Mito paper), plus the older ones (for Matt's paper)

Let's first get yours synced over because that's easy but note that there will be two different versions of analysis, backend, profiles and we need some appropriate nomenclature. What do you recommend?

- analysis, backend, profiles
- analysis_mito, backend_mito, profiles_mito

**Reply by ErinWeisbart** [reply]
I like using the standard `analysis`, `backend`, and `profiles` for the traditional versions made for the Tegtmeyer paper.

I'm not against using the `_mito` suffix for the ones for the Mito paper. An alternative that I can think of is making a `publication_data` workspace folder (like we did for PERISCOPE e.g. s3://cellpainting-gallery/cpg0021-periscope/broad/workspace/publication_data/2022_PERISCOPE/) and putting the `analysis_mito` etc in the `publication_data` subfolder since the mito profiles are quite different from most morphological profiles in the CPG in that they were from a single channel with a limited set of measurements.

Thoughts?

**Reply by shntnu** [reply]
I like the `publication_data` idea!

**Reply by ErinWeisbart** [reply]
It sounds like y'all just came to the conclusion that we won't be including the Mito-specific reanalysis in the upcoming Mito paper as the results were inconclusive. Let me know if you think we should still upload the Mito-specific analyses somewhere?

**Reply by shntnu** [reply]
We concluded, yes, we should make it public. I've been using `s5cmd` and `rclone` to sync and it's super fast (I use the latter when files sizes are too large, because s5cmd can't handle it)

https://github.com/jump-cellpainting/datasets-private/issues/106#issue-2779097593

**Reply by ErinWeisbart** [reply]
Happy to transfer the mito files, just need to figure out where - do we still think they fit well enough in `publication_data`. They won't have been included in the publication but were generated for the sake of a publication?

**Reply by shntnu** [reply]
> do we still think they fit well enough in `publication_data`.

Yes

**Reply by ErinWeisbart** [reply]
Great. Would the subfolder be `2025_Haghighi`?

**Reply by shntnu** [reply]
> Great. Would the subfolder be `2025_Haghighi`?

2025_Haghighi_McPhie

**Reply by ErinWeisbart** [reply]
The Mito-specific `analysis` (that I generated) is now in `cpg0038-tegtmeyer-neuropainting/broad/workspace/publication_data/2025_Haghighi_McPhie/`.

Johan (who analyzed the `analysis` data) used the raw single cell readouts so he did not create `backend` or `profiles`.

**Comment by ErinWeisbart** [comment]
Did final cleanup of backends/profile folders today so I think this is all good to go and will shortly be listed in README #137

**Comment by shntnu** [comment]
> * The Contributor will use provided credentials to upload data to the staging bucket, following [instructions](https://github.com/broadinstitute/cytoskel/blob/main/cytoskel/docs/access_cpg_staging.md) on accessing and using the credentials.
> * [ ]  The Champion will verify data structure and give the go-ahead to transfer from the staging bucket to the production bucket.
> * [ ]  A Maintainer will perform the transfer using an automated process with AWS S3 Batch.
> * [ ]  A Maintainer will disable the staging prefix, delete the data from staging, and delete credentials.

#137 was merged so I will check these off above

@mtegtmey - the single cell profiles are in cpg0038. The Cell Painting Gallery [docs](https://broadinstitute.github.io/cellpainting-gallery/data_structure.html) should be sufficient to find them. 

Once you do, can you please update the landing page of https://github.com/broadinstitute/neuropainting to point to this discussion + Cell Painting Gallery, so it is easy for others to find the data?
